homeassistant/components/komfovent/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for Komfovent integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+import komfovent_api
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER = "user"
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_USERNAME, default="user"): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+ERRORS_MAP = {
+    komfovent_api.KomfoventConnectionResult.NOT_FOUND: "cannot_connect",
+    komfovent_api.KomfoventConnectionResult.UNAUTHORISED: "invalid_auth",
+    komfovent_api.KomfoventConnectionResult.INVALID_INPUT: "invalid_input",
+}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Komfovent."""
+
+    VERSION = 1
+
+    def __return_error(
+        self, result: komfovent_api.KomfoventConnectionResult
+    ) -> FlowResult:
+        return self.async_show_form(
+            step_id=STEP_USER,
+            data_schema=STEP_USER_DATA_SCHEMA,
+            errors={"base": ERRORS_MAP.get(result, "unknown")},
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
```

Comment:
```suggestion
        errors = {}
        if user_input is not None:
            (Do the validation part here and fill errors if there are errors)
        return self.async_show_form(
            step_id=STEP_USER, data_schema=STEP_USER_DATA_SCHEMA, errors=errors
        )
```
I would suggest flipping it around so you only have 1 call where you actually create the form.

Reply:
I can do it if you insist, but this code is very sequential and does the checks inside the dependency, that's why it has a `KomfoventConnectionResult.INVALID_INPUT: "invalid_input",` mapping, so I dont really see a need for it here. It's not like any of the inputs have strict requirements that i could validate against.  
  
Addressed all other comments

Reply:
I mean more like, you now have 3 places where you return a form, and it looks cleaner to reduce that to only one place. checkout the config flow in lastfm for an example. With validation part I mean the get credentials and get settings calls you do here as well.

Reply:
I took a look at lastFM integration, and this approach still makes little sense to me. What's the point of collecting to an array and adding a bunch of IFs to just save on return statements? IMO early returns are much more readable and take less lines than this mess.  
  
If it is a hard requirement from you, please state so and I will implement it, but if I can have a say, I would rather keep it as it is.

Reply:
It's not a hard requirement. :)